evalml/automl/automl_search.py

```diff
@@ -1006,6 +1009,24 @@ def search(self, show_iteration_plot=True):
                 f"Best pipeline {self.objective.name}: {best_pipeline['validation_score']:3f}"
             )
         self._searched = True
+        if self.search_iteration_plot is not None:
+            # go = import_or_raise(
+            #     "plotly.graph_objects",
+            #     error_msg="Cannot find dependency plotly.graph_objects",
+            # )
+            # title = "Pipeline Search: Iteration vs. {}<br><sub>Gray marker indicates the score at current iteration</sub>".format(
+            #     self.objective.name
+            # )
+            # layout = {
+            #     "title": title,
+            #     "xaxis": {"title": "Iteration", "rangemode": "tozero"},
+            #     "yaxis": {"title": "Score"},
+            # }
+            if not show_iteration_plot:
+                self.search_iteration_plot = self.plot.search_iteration_plot(
+                    interactive_plot=show_iteration_plot
+                )
+                return self.search_iteration_plot
```

Comment:
Do we need to return this? Does it not render in the docs without it?

Reply:
From what I remember, without returning the `self.search_iteration_plot`, the docs don't render.

I could use figure.show() but that causes tests to take a really long time. (https://github.com/alteryx/evalml/runs/7100924419?check_suite_focus=true)

TL;DR: the automl tests was taking over an hour when it usually takes like 20-25 minutes when I used figure.show(). When I ran the test locally, figure.show() caused every test from automl to open a web browser tab and show me the graph, multiple times (before I stopped it manually safari had like 30+ tabs of just graphs). I suspect that's the reason why it takes so long (for whatever reason). 

I tried to mess around with figure.show() parameters, but couldn't figure out a way to render the graph so that:
1. It's interactive (and not like a jpg or png of the graph)
2. It doesn't make the local test open up a browser every time it runs `automl.search()`
3. Do both of the above and actually render when I build the doc using sphinx (sometimes it loads locally, then I check and it gives an error message when I upload it to github).

Of course there's a high chance I'm missing something pretty obvious (maybe I could try making it so that it tries to use your web-browser to render it, then it doesn't?).